takahe/settings.py

```diff
@@ -0,0 +1,242 @@
+import secrets
+import urllib.parse
+from pathlib import Path
+from typing import List, Literal, Optional, Union
+
+import dj_database_url
+import sentry_sdk
+from pydantic import AnyUrl, BaseSettings, EmailStr, Field, PostgresDsn
+from sentry_sdk.integrations.django import DjangoIntegration
+
+BASE_DIR = Path(__file__).resolve().parent.parent
+
+
+def as_bool(v: Optional[Union[str, List[str]]]):
+    if v is None:
+        return False
+
+    if isinstance(v, str):
+        v = [v]
+
+    return v[0].lower() in ("true", "yes", "t", "1")
+
+
+class Settings(BaseSettings):
+    """
+    Pydantic-powered settings, to provide consistent error messages, strong
+    typing, consistent prefixes, .venv support, etc.
+    """
+
+    #: The default database.
+    DATABASE_URL: PostgresDsn
+    #: The currently running environment, used for things such as sentry
+    #: error reporting.
+    ENVIRONMENT: Literal["dev", "prod", "test"] = "dev"
+    #: Should django run in debug mode?
+    DEBUG: bool = False
+    #: Set a secret key used for signing values such as sessions. Randomized
+    #: by default, so you'll logout everytime the process restarts.
+    SECRET_KEY: str = Field(default_factory=lambda: secrets.token_hex(128))
+    #: Set a secret key used to protect the stator. Randomized by default.
+    STATOR_TOKEN: str = Field(default_factory=lambda: secrets.token_hex(128))
+
+    #: If set, a list of allowed values for the HOST header. The default value
+    #: of '*' means any host will be accepted.
+    ALLOWED_HOSTS: List[str] = Field(default_factory=lambda: ["*"])
+    #: If set, a list of hosts to accept for CORS.
+    CORS_HOSTS: List[str] = Field(default_factory=list)
+    #: If set, a list of hosts to accept for CSRF.
+    CSRF_HOSTS: List[str] = Field(default_factory=list)
+    #: If enabled, trust the HTTP_X_FORWARDED_FOR header.
+    USE_PROXY_HEADERS: bool = False
+
+    #: An optional Sentry DSN for error reporting.
+    SENTRY_DSN: Optional[str] = None
+
+    #: Fallback domain for links.
+    MAIN_DOMAIN: str = "example.com"
+
+    EMAIL_DSN: AnyUrl = "console://localhost"
+    EMAIL_FROM: EmailStr = "test@example.com"
+    AUTO_ADMIN_EMAIL: Optional[EmailStr] = None
+    ERROR_EMAILS: Optional[List[EmailStr]] = None
+
+    MEDIA_URL: str = "/media/"
+    MEDIA_ROOT: str = str(BASE_DIR / "MEDIA")
+    MEDIA_BACKEND: Optional[AnyUrl] = None
+
+    class Config:
+        env_prefix = "TAKAHE_"
+        env_file = str(BASE_DIR / ".env")
+        env_file_encoding = "utf-8"
+        # Case sensitivity doesn't work on Windows, so might as well be
+        # consistent from the get-go.
+        case_sensitive = False
+
+
+CONFIG = Settings()
```

Comment:
The internal database-driven settings system is already called Config, so this will need a different name